source/common/config/well_known_names.cc

```diff
@@ -132,6 +134,13 @@ TagNameValues::TagNameValues() {
   // internal state of the regex which performs better.
   addRe2(HTTP_CONN_MANAGER_PREFIX, R"(^listener\..*?\.http\.((<TAG_VALUE>)\.))", ".http.");
 
+  // Extract ext_authz stat_prefix field
+  // cluster.[<cluster>.]ext_authz.[<ext_authz_prefix>.]*
+  // http.[<http_conn_mgr_prefix>.]ext_authz.[<ext_authz_prefix>.]*
+  if (Runtime::runtimeFeatureEnabled("envoy.restart_features.ext_authz_prefix_handling")) {
+    addRe2(EXT_AUTHZ_PREFIX, R"(^(?:cluster|http)\.(?:[^\.]+\.)?ext_authz\.((<TAG_VALUE>)\.))");
```

Comment:
From a performance perspective, the parenthesized condition at the start of the regex blocks an optimization, forcing this regex to be evaluated on every stat.

The optimization looks for regexes that start with `^foo.` and puts those in a map from "foo"->list(matchers), and only considers those matchers on stats starting with "foo".

I think you can easily rewrite the above regex into two of them that will enable this optimization. WDYT?